scripts/zones/Bibiki_Bay/mobs/Shen.lua

```diff
@@ -2,36 +2,29 @@
 -- Area: Bibiki Bay
 --  Mob: Shen
 -----------------------------------
+local ID = require("scripts/zones/Bibiki_Bay/IDs")
+require("scripts/mixins/families/uragnite")
+-----------------------------------
 local entity = {}
 
+entity.onMobSpawn = function(mob)
+end
+
 entity.onMobFight = function(mob, target)
-    if (mob:getBattleTime() % 45 == 0) then
-        local mobId = mob:getID()
-        for i = 1, 2 do
-            if (not GetMobByID(mobId+i):isSpawned()) then
-                SpawnMob(mobId+i):updateEnmity(target)
+    local spawnFiltrate = function()
```

Comment:
Is there a reason for defining a function here only to call it right below?  Seems redundant

Reply:
I was told to make this a local function. I must have been confused lol. Should I keep it local and just move it outside of onMobFight?

Reply:
You're right, and I apologise for my original review.  I meant more that it shouldn't be attached to the object itself, since its unnecessary.  In this case, the function is only used in one spot, so there's really not a need to define a function.

```
    local spawnFiltrate = function()
        if mob:getAnimationSub() == 1 and mob:getHP() > 0 then
            for i = 1, 2 do
                if not GetMobByID(mob:getID()+i):isSpawned() then
                    SpawnMob(mob:getID()+i):updateEnmity(target)
                end
            end
        end
    end

    spawnFiltrate()
```
That's the same as saying:
```
      if mob:getAnimationSub() == 1 and mob:getHP() > 0 then
          for i = 1, 2 do
              if not GetMobByID(mob:getID()+i):isSpawned() then
                  SpawnMob(mob:getID()+i):updateEnmity(target)
              end
          end
      end
```

Reply:
Oh! Most definitely. I'll try and employ this more in the future. Thank you